test/ref/unique.cpp

```diff
@@ -0,0 +1,338 @@
+    auto r    = mm->add_instruction(op, data);
+
+    auto r0 = mm->add_instruction(migraphx::make_op("get_tuple_elem", {{"index", 0}}), r);
+    auto r1 = mm->add_instruction(migraphx::make_op("get_tuple_elem", {{"index", 1}}), r);
+    auto r2 = mm->add_instruction(migraphx::make_op("get_tuple_elem", {{"index", 2}}), r);
+    auto r3 = mm->add_instruction(migraphx::make_op("get_tuple_elem", {{"index", 3}}), r);
+    mm->add_return({r0, r1, r2, r3});
+    return p;
+};
+
+template <typename T>
+auto run_program(T& data,
+                 const migraphx::shape& data_shape,
+                 int sorted,
+                 int64_t axis = std::numeric_limits<int64_t>::max())
+{
+    auto p = create_program(data_shape, sorted, axis);
+    p.compile(migraphx::make_target("ref"));
+    migraphx::parameter_map pp;
+    pp["X"]   = migraphx::argument(data_shape, data.data());
+    auto rets = p.eval(pp);
+    std::vector<typename std::remove_reference_t<decltype(data)>::value_type> y;
+    rets[0].visit([&](auto v) { y.assign(v.begin(), v.end()); });
+    std::vector<int64_t> y_idx;
+    rets[1].visit([&](auto v) { y_idx.assign(v.begin(), v.end()); });
+    std::vector<int64_t> x_rev_idx;
+    rets[2].visit([&](auto v) { x_rev_idx.assign(v.begin(), v.end()); });
+    std::vector<int64_t> y_ct;
+    rets[3].visit([&](auto v) { y_ct.assign(v.begin(), v.end()); });
+
+    return std::make_tuple(y, y_idx, x_rev_idx, y_ct);
+}
+} // namespace
+
+TEST_CASE(unique_test)
+{
+    // case 1  sorted. single entry
+    {
+        std::vector<int> data    = {2};
+        int64_t axis             = 0;
+        int64_t sorted           = 1;
+        std::vector<size_t> lens = {1};
+        migraphx::shape data_shape{migraphx::shape::int32_type, lens};
+        const auto& [y, idx, x_rev, ct] = run_program(data, data_shape, sorted, axis);
+
+        std::vector<int> gold_val = {2};
+        EXPECT(y == gold_val);
+
+        std::vector<int64_t> gold_y_idx = {0};
+        EXPECT(idx == gold_y_idx);
+
+        std::vector<int64_t> gold_x_rev = {0};
+        EXPECT(x_rev == gold_x_rev);
+
+        std::vector<int64_t> gold_ct = {1};
+        EXPECT(ct == gold_ct);
+    }
+
+    // case 1  unsorted. single entry
+    {
+        std::vector<float> data  = {3.33};
+        int64_t axis             = -1;
+        int64_t sorted           = 0;
+        std::vector<size_t> lens = {1};
+        migraphx::shape data_shape{migraphx::shape::float_type, lens};
+        const auto& [y, idx, x_rev, ct] = run_program(data, data_shape, sorted, axis);
+
+        std::vector<float> gold_val = {3.33};
+        EXPECT(y == gold_val);
+
+        std::vector<int64_t> gold_y_idx = {0};
+        EXPECT(idx == gold_y_idx);
+
+        std::vector<int64_t> gold_x_rev = {0};
+        EXPECT(x_rev == gold_x_rev);
+
+        std::vector<int64_t> gold_ct = {1};
+        EXPECT(ct == gold_ct);
+    }
+
+    // case 2  sorted. all unique input..
```

Comment:
can we create different test cases for all those cases ?

Reply:
You mean separate them out within this file? Thanks.

Reply:
yes separate `TEST_CASE` for each case.

Reply:
Done.